app/javascript/mastodon/features/list_timeline/index.js

```diff
@@ -138,15 +139,22 @@ class ListTimeline extends React.PureComponent {
   handleRepliesPolicyChange = ({ target }) => {
     const { dispatch } = this.props;
     const { id } = this.props.params;
-    dispatch(updateList(id, undefined, false, target.value));
-  };
+    dispatch(updateList(id, undefined, false, undefined, target.value));
+  }
+
+  onExclusiveToggle = () => {
+    const { dispatch, list } = this.props;
+    const { id } = this.props.params;
+    dispatch(updateList(id, undefined, false, !list.get('is_exclusive'), undefined));
+  }
```

Comment:
This introduces a race condition, where when the toggle is clicked twice in rapid succession or the database is slow for whatever reason, the value send could be different from the state of the toggle.
I would rather have the state of the toggle send here, so it always matches.

```suggestion
  onExclusiveToggle = e => {
    const { dispatch } = this.props;
    const { id } = this.props.params;
    dispatch(updateList(id, undefined, false, e.target.checked, undefined));
  }
```